src/Microsoft.Data/BaseColumn.cs

```diff
@@ -128,11 +140,23 @@ public void SetName(string newName, DataFrame dataFrame = null)
         /// <param name="upper">Maximum value. All values above this threshold will be set to it</param>
         public virtual BaseColumn Clip<U>(U lower, U upper) => throw new NotImplementedException();
 
+        /// <summary>
+        /// Returns a new column filtered by the lower and upper bounds
+        /// </summary>
+        /// <typeparam name="U"></typeparam>
+        /// <param name="lower"></param>
+        /// <param name="upper"></param>
+        public virtual BaseColumn Filter<U>(U lower, U upper) => throw new NotImplementedException();
+
         /// <summary>
         /// Determines if the column is of a numeric type
         /// </summary>
         public virtual bool IsNumericColumn() => false;
 
+        public virtual double Mean() => throw new NotImplementedException();
```

Comment:
Are the xml-comments intentionally left out?

Reply:
Fixed